pkg/resource/deploy/source_eval.go

```diff
@@ -595,7 +607,11 @@ func (rm *resmon) StreamInvoke(
 	}
 
 	args, err := plugin.UnmarshalProperties(
-		req.GetArgs(), plugin.MarshalOptions{Label: label, KeepUnknowns: true})
+		req.GetArgs(), plugin.MarshalOptions{
+			Label:        label,
+			KeepUnknowns: true,
+			KeepSecrets:  true,
```

Comment:
Why do we need to set `KeepSecrets` now?

Reply:
This was an oversight from the original changes that added either `StreamInvoke` or secrets support (probably the former). This flag should always have been set.